### ISSUE TITLE
Hidden range in nn.MaxPool2d

### ISSUE THREAD

**Issue by PhyllisJi** [pytorch/pytorch#103423]
### Describe the bug

`Hidden range of padding parameter in nn.MaxPool2d`

`pad should be at most half of kernel size, but got pad=2 and kernel_size=2`

### Code

```py
import torch
from torch import nn


class lenet(nn.Module):
    def __init__(self):
        super(lenet, self).__init__()
        self.conv = nn.Conv2d(in_channels=3, out_channels=16, kernel_size=2, padding=2, stride=1)
        self.pool = nn.MaxPool2d(padding=2, kernel_size=2)

    def forward(self, x):
        # 1st block
        x = self.conv(x)
        x = self.pool(x)

        return x


if __name__ == '__main__':
    net = lenet()
```

### Version
PyTorch version: 2.0.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.6 (arm64)
GCC version: Could not collect
Clang version: 14.0.0 (clang-1400.0.29.102)
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.10 (main, Mar 21 2023, 13:41:05) [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-12.6-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] torch==2.0.0
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.15.1
[conda] numpy                     1.23.5          py310hb93e574_0  
[conda] numpy-base                1.23.5          py310haf87e8b_0  
[conda] torch                     2.0.0                    pypi_0    pypi
[conda] torchsummary              1.5.1                    pypi_0    pypi
[conda] torchvision               0.15.1                   pypi_0    pypi

**Comment by PhyllisJi** [comment]
> `pad should be at most half of kernel size, but got pad=2 and kernel_size=2`

I mean that the official document does not emphasize the dependency “pad should be at most half of kernel size, but got pad=2 and kernel_size=2”
